File: Python3/554. Brick Wall.py
class Solution:
    def leastBricks(self, wall):
        """
        :type wall: List[List[int]]
        :rtype: int
        """
        edges = collections.defaultdict(int)
        length = sum(wall[0])
        for row in wall:
            total = 0
            for brick in row:
                total += brick
                if total < length:
                    edges[total] += 1
        if not edges:
            return len(wall)
        return len(wall) - max(edges.values())

File: Golang/554. Brick Wall.go
func leastBricks(wall [][]int) int {
    hashtable := make(map[int]int, 100)
    for _, row := range wall {
        total := 0
        for _, brick := range row[: len(row)-1] {
            total += brick
            hashtable[total]++
        }
    }
    ans := 0
    for k := range hashtable {
        if hashtable[k] > ans {
            ans = hashtable[k]
        }
    }
    return len(wall) - ans
}
